context.go

```diff
@@ -293,24 +303,85 @@ func (c *context) Scheme() string {
 	return "http"
 }
 
+func (c *context) SchemeForwarded() *Forwarded {
+	// Parse and get "Forwarded" header.
+	// See : https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+	if scheme := c.request.Header.Get(HeaderForwarded); scheme != "" {
+		f, err := c.parseForwarded(scheme)
+		if err != nil {
+			return nil
+		}
+		return &f
+	}
+	return nil
+}
+
+func (c *context) parseForwarded(input string) (Forwarded, error) {
+	forwarded := Forwarded{}
+	entries := strings.Split(input, ",")
+
+	for _, entry := range entries {
+		entry = strings.TrimSpace(entry)
+		pairs := strings.Split(entry, ";")
+
+		for _, pair := range pairs {
+			parts := strings.SplitN(pair, "=", 2)
+			if len(parts) != 2 {
+				return forwarded, fmt.Errorf("invalid pair: %s", pair)
+			}
+
+			key := strings.TrimSpace(parts[0])
+			value, err := url.QueryUnescape(strings.TrimSpace(parts[1]))
```

Comment:
Is `url.QueryUnescape` correct by rfc7230 section-3.2.6? 

RFC says that `value` is defined as:
> forwarded-pair = token "=" value
    value  = token / quoted-string
     token = <Defined in [[RFC7230], Section 3.2.6](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6)>
     quoted-string = <Defined in [[RFC7230], Section 3.2.6](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6)>